system/lib/wasmfs/js_impl_backend.h

```diff
@@ -44,7 +44,7 @@
 // For a simple example, see js_file_backend.cpp and library_wasmfs_js_file.js
 //
 
-using js_index_t = uint32_t;
+using js_index_t = uintptr_t;
```

Comment:
Why did these change? IIRC, these are indices into JS arrays and can still never have more than 32 useful bits.

Reply:
We have `static_assert(sizeof(backend_t) == sizeof(js_index_t)` in the code and `backend_t` is a pointer.. so I assumed it was necessary.  Is it not?

Reply:
```
  js_index_t getBackendIndex() {                                                 
    static_assert(sizeof(backend_t) == sizeof(js_index_t), "TODO: wasm64");      
    return js_index_t(getBackend());                                                
  }                                                                              
                                                                                 
  js_index_t getFileIndex() {                                                    
    static_assert(sizeof(this) == sizeof(js_index_t), "TODO: wasm64");           
    return js_index_t(this);                                                     
  }   
  ```

The simplest way to make that work was just to make js_index_t pointer sized.

Reply:
@kripken, is that what you intended when you wrote that? It looks like I misremembered `js_index_t` being an index into a JS array.

Reply:
I guess we could use an atomic i32 to map backends and files to uint32_t?  Something like `static _Atomic uint32_t g_last_backend_index = 0;`  that get increments in the backend constructor?

The cost we could one extra uint32_t in every file and every backend.. whereas the cost of widening js_index_t to uintptr_t is zero for wasm32.. but extra wrapper code on wasm64.

Reply:
Hmm, good catch, it does look like we've "overloaded" `js_index_t` to mean either a JS index, or a backend index. But a backend index is a pointer, so those have different sizes in wasm64.

I think it would be good to fix that. Basically all the places declaring `js_index_t backend` should become `void* backend` (or perhaps a new type for a backend pointer).

Reply:
js_index_t seems to have two different uses:

1. Index pointing to a backend
2. Index pointing to a file within a given backend

In both cases the C side is assuming that the js indexes map to pointers on the C/C++ side.  Is this a problem that needs fixing?

Reply:
OK I'm making a separate PR to avoid passing pointers as handles.